library/ssl_tls.c

```diff
@@ -7222,10 +7222,25 @@ static int ssl_parse_certificate_chain( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_SSL_KEEP_PEER_CERTIFICATE */
         switch( ret )
         {
-            case 0: /*ok*/
+            case 0: /* ok */
+                mbedtls_platform_random_delay();
+                if( ret != 0 )
+                {
+                    alert = MBEDTLS_SSL_ALERT_MSG_INTERNAL_ERROR;
+                    ret = MBEDTLS_ERR_PLATFORM_FAULT_DETECTED;
+                    goto crt_parse_der_failed;
+                }
+                break;
```

Comment:
@piotr-now , the flow didn't have "break" before your fix. What was the reason it always went to default ?

Reply:
Previously there was one `brake` for `case 0:` and `case MBEDTLS_ERR_X509_UNKNOWN_SIG_ALG + MBEDTLS_ERR_OID_NOT_FOUND:` because both of them are correct and without any additional operations. If the switch fell into the `case 0:` then it was called a brake from the second case. After adding an additional check, the program need to be brake before checking in the second case.